### PR TITLE
Upgrade to ESLint 8

### PR DESCRIPTION
**Pull request by Droniu**
I want to merge this change because it upgrades ESLint to major version 8. We will also now follow recommended ESLint configs - `standard-with-typescript` and `plugin:react/recommended`. Rules conflicting with prettier have been turned off.
This PR fixes the following rules:
```
@typescript-eslint/ban-tslint-comment
@typescript-eslint/consistent-indexed-object-style
@typescript-eslint/no-unnecessary-boolean-literal-compare
@typescript-eslint/prefer-includes
@typescript-eslint/prefer-readonly
@typescript-eslint/prefer-ts-expect-error
eol-last
no-unneeded-ternary
no-useless-return
operator-linebreak
react/jsx-no-target-blank
react/no-unknown-property
spaced-comment
no-multiple-empty-lines
no-trailing-spaces
padded-blocks
simple-import-sort/exports
simple-import-sort/imports
```

Also `eslint-nibble` has been added for easier tracking of migration to new rules. It's available as `npm run lint-check`

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `data-test-id` are added for new elements
4. [ ] The changes are tested in Chrome/Firefox/Safari browsers and in light/dark mode
5. [ ] Your code works with the latest stable version of the core

### Test environment config

API_URI=https://automation-dashboard.staging.saleor.cloud/graphql/
APPS_MARKETPLACE_API_URI=https://apps.staging.saleor.io/api/v2/saleor-apps

### Do you want to run more stable tests?

To run all tests, just select the stable checkbox. To speed up tests, increase the number of containers. Tests will be re-run only when the "run e2e" label is added.

1. [x] stable
2. [ ] giftCard
3. [ ] category
4. [ ] collection
5. [ ] attribute
6. [ ] productType
7. [ ] shipping
8. [ ] customer
9. [ ] permissions
10. [ ] menuNavigation
11. [ ] pages
12. [ ] sales
13. [ ] vouchers
14. [ ] homePage
15. [ ] login
16. [ ] orders
17. [ ] products
18. [ ] app
19. [ ] plugins
20. [ ] translations
21. [ ] navigation
22. [ ] variants
23. [ ] payments

CONTAINERS=4

### CODE REVIEW

File: src/orders/components/OrderTransaction/components/TransactionEvents/components/PspReferenceLink.tsx [hunk 1]
```diff
@@ -26,7 +26,7 @@ export const PspReferenceLink: React.FC<PspRerefenceLinkProps> = ({
       <a
         href={href}
         target="_blank"
-        rel="noopener noreferer"
+        rel="noopener noreferer noreferrer"
```
Comment (poulch) [inline src/orders/components/OrderTransaction/components/TransactionEvents/components/PspReferenceLink.tsx:29]:
```suggestion
        rel="noopener noreferrer"
```

File: src/components/Filter/FilterKeyValueField.tsx [hunk 2]
```diff
@@ -34,8 +34,8 @@ const useStyles = makeStyles(
   }),
   { name: "FilterKeyValueField" },
 );
-
-const getUpdateArrayFn = <T,>(key: "key" | "value") => (
+// @eslint-ignore-next-line
+const getUpdateArrayFn = <T,>(key: "key" | "value") => ( // eslint-disable-line
```
Comment (poulch) [inline src/components/Filter/FilterKeyValueField.tsx:38]: question: do you need both comments?

File: package.json [hunk 3]
```diff
@@ -321,6 +326,7 @@
     "test": "jest src/",
     "test:watch": "jest --watch src/",
     "lint": "eslint \"src/**/*.@(tsx|ts|jsx|js)\" --fix",
+    "lint-check": "eslint-nibble \"src/**/*.@(tsx|ts|jsx|js)\"",
```
Comment (krzysztofzuraw) [inline package.json:328]:
```suggestion
    "lint:check-progress": "eslint-nibble \"src/**/*.@(tsx|ts|jsx|js)\"",
```

File: src/discounts/components/DiscountCountrySelectDialog/DiscountCountrySelectDialog.tsx [hunk 4]
```diff
@@ -95,7 +95,7 @@ const DiscountCountrySelectDialog: React.FC<
                   name="query"
                   value={data.query}
                   onChange={event =>
-                    change(event /* TO BE CHECKED: () => fetch(data.query)*/)
+                    change(event /* TO BE CHECKED: () => fetch(data.query) */)
```
Comment (krzysztofzuraw) [inline src/discounts/components/DiscountCountrySelectDialog/DiscountCountrySelectDialog.tsx:98]: **suggestion**: can you remove `TO BE CHECKED`?

**Review by krzysztofzuraw — APPROVED** [review]

**Review by poulch — DISMISSED** [review]

**Review by krzysztofzuraw — APPROVED** [review]

**Review by andrzejewsky — APPROVED** [review]

**Review by poulch — APPROVED** [review]

**Pull request merged**